Update docs about Automatic Prerendering (#7816)
zeit/next.js: packages/next/README.md

  </ul>
</details>

A custom `<Document>` is commonly used to augment your application's `<html>` and `<body>` tags.
This is necessary because Next.js pages skip the definition of the surrounding document's markup.

This allows you to support Server-Side Rendering for CSS-in-JS libraries like
[styled-components](/examples/with-styled-components) or [emotion](/examples/with-emotion).
Note, [styled-jsx](https://github.com/zeit/styled-jsx) is included in Next.js by default.

A custom `<Document>` can also include `getInitialProps` for expressing asynchronous server-rendering data requirements.

> **Note**: `<Document>`'s `getInitialProps` function is not called during client-side transitions,
> nor when a page is [automatically prerendered](#automatic-prerendering).

> **Note**: Make sure to check if `ctx.req` / `ctx.res` are defined in `getInitialProps`.
> These variables will be `undefined` when a page is being statically exported for `next export` or [automatic prerendering (static optimization)](#automatic-prerendering).

> To use a custom `<Document>`, must create a file at `./pages/_document.js` and extend the `Document` class:

Reviewer: ```suggestion
To use a custom `<Document>`, you create a file at `./pages/_document.js` and extend the `Document` class:
```

## Automatic Prerendering

> Next.js will automatically determine a page is static (and can be prerendered) if it has no has blocking data requirements.

Reviewer: ```suggestion
Next.js will automatically determine if a page is static (can be prerendered) and it has no blocking data requirements.
```

Author: How about

> Next.js automatically determines that a page is static (can be prerendered) if it has no has blocking data requirements.

?

Reviewer: sounds good, updated the suggestion

This determination is made by the absence of `getInitialProps` in both the page itself and a [custom `<App>`](#custom-app) (if present).

If `getInitialProps` is present, Next.js will not prerender the page.
Instead, Next.js will use its default behavior and render the page on-demand, per-request (meaning Server-Side Rendering).

> If `getInitialProps` is abset, Next.js will **statically optimize** your page automatically by prerendering it to static HTML.

Reviewer: ```suggestion
If `getInitialProps` is absent, Next.js will **statically optimize** your page automatically by prerendering it to static HTML.
```

This feature allows Next.js to emit hybrid applications that contain **both server-rendered and statically generated pages**.
This ensures Next.js always emits applications that are **fast by default**.

> **Note**: Statically generated pages are still reactive: Next.js will hydrate your application client-side to give it full interactivity.

This feature provides many benefits.
For example, optimized pages require no server-side computation and can be instantly streamed to the end-user from CDN locations.

> The result is an _ultra fast_ loading experience for your users.

Reviewer: ```suggestion
This results is an _ultra fast_ loading experience for your users.
```
